File: c++/AbstractFactoryPattern/AudiFactory.cpp
#include "stdafx.h"
#include "AudiFactory.h"

IEngine *AudiFactory::CreateEngine()
{
	return new AudiEngine();
}

IBody *AudiFactory::CreateBody()
{
	return new AudiBody();
}

IWheel *AudiFactory::CreateWheel()
{
	return new AudiWheel();
}

File: csharp/AbstractFactoryPattern/Factory/AudiFactory.cs

using AbstractFactoryPattern.ProductBody;
using AbstractFactoryPattern.ProductEngine;
using AbstractFactoryPattern.ProductWheel;
namespace AbstractFactoryPattern.Factory
{
    /// <summary>
    /// 具体的奥迪工厂类
    /// </summary>
    public class AudiFactory : ICarFactory
    {
        public IEngine CreateEngine()
        {
            return new AudiEngine();
        }

        public IWheel CreateWheel()
        {
            return new AudiWheel();
        }

        public IBody CreateBody()
        {
            return new AudiBody();
        }

    }
}
